dist/platforms/ubuntu/steps/activate.sh

```diff
@@ -4,76 +4,30 @@
 echo "Changing to \"$ACTIVATE_LICENSE_PATH\" directory."
 pushd "$ACTIVATE_LICENSE_PATH"
 
-if [[ -n "$UNITY_LICENSE" ]] || [[ -n "$UNITY_LICENSE_FILE" ]]; then
+if [[ -n "$UNITY_SERIAL" && -n "$UNITY_EMAIL" && -n "$UNITY_PASSWORD" ]]; then
   #
-  # PERSONAL LICENSE MODE
-  #
-  # This will activate Unity, using a license file
-  #
-  # Note that this is the ONLY WAY for PERSONAL LICENSES in 2020.
-  #   * See for more details: https://gitlab.com/gableroux/unity3d-gitlab-ci-example/issues/5#note_72815478
-  #
-  # The license file can be acquired using `webbertakken/request-manual-activation-file` action.
-  echo "Requesting activation (personal license)"
-
-  # Set the license file path
-  FILE_PATH=UnityLicenseFile.ulf
-
-  if [[ -n "$UNITY_LICENSE" ]]; then
-    # Copy license file from Github variables
-    echo "$UNITY_LICENSE" | tr -d '\r' > $FILE_PATH
-  elif [[ -n "$UNITY_LICENSE_FILE" ]]; then
-    # Copy license file from file system
-    cat "$UNITY_LICENSE_FILE" | tr -d '\r' > $FILE_PATH
-  fi
-
-  # Activate license
-  ACTIVATION_OUTPUT=$(unity-editor \
-      -logFile /dev/stdout \
-      -quit \
-      -manualLicenseFile $FILE_PATH)
-
-  # Store the exit code from the verify command
-  UNITY_EXIT_CODE=$?
-
-  # The exit code for personal activation is always 1;
-  # Determine whether activation was successful.
-  #
-  # Successful output should include the following:
-  #
-  #   "LICENSE SYSTEM [2020120 18:51:20] Next license update check is after 2019-11-25T18:23:38"
-  #
-  ACTIVATION_SUCCESSFUL=$(echo $ACTIVATION_OUTPUT | grep 'Next license update check is after' | wc -l)
-
-  # Set exit code to 0 if activation was successful
-  if [[ $ACTIVATION_SUCCESSFUL -eq 1 ]]; then
-    UNITY_EXIT_CODE=0
-  fi;
-
-  # Remove license file
-  rm -f $FILE_PATH
-
-elif [[ -n "$UNITY_SERIAL" && -n "$UNITY_EMAIL" && -n "$UNITY_PASSWORD" ]]; then
-  #
-  # PROFESSIONAL (SERIAL) LICENSE MODE
+  # SERIAL LICENSE MODE
   #
   # This will activate unity, using the activating process.
   #
-  # Note: This is the preferred way for PROFESSIONAL LICENSES.
-  #
-  echo "Requesting activation (professional license)"
+  echo "Requesting activation"
 
   # Activate license
   unity-editor \
     -logFile /dev/stdout \
     -quit \
     -serial "$UNITY_SERIAL" \
     -username "$UNITY_EMAIL" \
-    -password "$UNITY_PASSWORD"
+    -password "$UNITY_PASSWORD" \
+    -projectPath "/BlankProject"
```

Comment:
Are you sure about this change? Does it work for all versions? We separated between the two methods because of necessity back then. A single flag can make a rigorous difference in the Unity command line and show entirely different behaviour at times.

If you're sure about this, I welcome the simplification :)

Reply:
Yea so the reason I switched to this is it seems that activating with serial likes having a real project path provided. I was getting some odd errors without it. I think when it was doing license file activation it didn't care since it was probably just moving the ulf to the right folder rather than actually spinning up Unity for the licensing client which is what happens now since manual ulf doesn't seem to work anymore unfortunately.